spec/highlighting.spec.js

```diff
@@ -54,4 +84,400 @@ describe('Highlighting', function () {
+</span>
+ Make The <br> World Go Round`)
+      expect(this.getHtml()).toEqual(expectedHtml)
+      expect(this.extract()).toEqual(expectedRanges)
+    })
+
+    it('can handle intersecting highlights', () => {
+      this.highlightRange('first', 0, 3)
+      this.highlightRange('second', 3, 7)
+      this.highlightRange('third', 4, 6)
+      const expectedRanges = {
+        first: {
+          text: 'Peo',
+          start: 0,
+          end: 3
+        },
+        second: {
+          text: 'ple ',
+          start: 3,
+          end: 7
+        },
+        third: {
+          text: 'le',
+          start: 4,
+          end: 6
+        }
+      }
+      const expectedHtml = this.formatHtml(`<span class="highlight-comment" data-word-id="first" data-editable="ui-unwrap" data-highlight="comment">Peo</span>
+<span class="highlight-comment" data-word-id="second" data-editable="ui-unwrap" data-highlight="comment">p
+<span class="highlight-comment" data-word-id="third" data-editable="ui-unwrap" data-highlight="comment">le</span>
+ </span>
+Make The <br> World Go Round`)
+      expect(this.getHtml()).toEqual(expectedHtml)
+      expect(this.extract()).toEqual(expectedRanges)
+    })
+
+    it('can handle highlights containing break tags', () => {
+      this.highlightRange('first', 11, 22)
+      const expectedRanges = {
+        first: {
+          text: ' The \nWorld',
+          start: 11,
+          end: 22
+        }
+      }
+      const expectedHtml = this.formatHtml(`People Make
+<span class="highlight-comment" data-word-id="first" data-editable="ui-unwrap" data-highlight="comment"> The <br> World</span>
+ Go Round`)
+
+      expect(this.extract()).toEqual(expectedRanges)
+      expect(this.getHtml()).toEqual(expectedHtml)
+
+    })
+
+    it('can handle identical ranges', () => {
+      this.highlightRange('first', 11, 22)
+      this.highlightRange('second', 11, 22)
+      const expectedRanges = {
+        first: {
+          text: ' The \nWorld',
+          start: 11,
+          end: 22
+        },
+        second: {
+          text: ' The \nWorld',
+          start: 11,
+          end: 22
+        }
+      }
+      const expectedHtml = this.formatHtml(`People Make
+<span class="highlight-comment" data-word-id="second" data-editable="ui-unwrap" data-highlight="comment">
+<span class="highlight-comment" data-word-id="first" data-editable="ui-unwrap" data-highlight="comment"> The <br> World</span>
+</span>
+ Go Round`)
+
+
+      expect(this.getHtml()).toEqual(expectedHtml)
+      expect(this.extract()).toEqual(expectedRanges)
+
+    })
+
+    it('will update any existing range found under `highlightId` aka upsert', () => {
+      this.highlightRange('first', 11, 22)
+      this.highlightRange('first', 8, 9)
+      const expectedRanges = {
+        first: {
+          text: 'a',
+          start: 8,
+          end: 9
+        }
+      }
+      const expectedHtml = this.formatHtml(`People M
```

Comment:
Two closing spans. Why?

Reply:
This was my typo. The test was still passing because of [this](https://github.com/livingdocsIO/editable.js/pull/164/files#diff-3990f96ac4ef1792695bda53e7fd3ecfR30)